configure.ac

```diff
@@ -610,10 +610,7 @@ AS_IF([test "`cd $srcdir && pwd`" != "`pwd`"],
      ['-I$(srcdir) -I$(top_builddir) -I$(top_srcdir) $(BOOST_CPPFLAGS)'])],
   [AC_SUBST([AM_CPPFLAGS],
      ['-I$(top_builddir) $(BOOST_CPPFLAGS)'])])
-
-#  Check if we can assume whether All Warnings Are Really Errors
-AS_IF([test xyes = x$ac_cv_cxx_compiler_gnu],
-  [gcc_AWARE=-Werror])
```

Comment:
Treating `gcc` compiler warnings as errors is project policy.  If need be, you can override this on an error-specific basis with
```
./configure CXX_FLAGS=-Wno-<name-of-error-to-ignore>
```